core/uploader.go

```diff
@@ -88,3 +96,54 @@ func Upload(input io.Reader, outputURI string, waitBetweenWrites, writeTimeout t
 
 	return nil
 }
+
+func extractThumb(session drivers.OSSession) {
+	presigned, err := session.Presign("", 5*time.Minute)
+	if err != nil {
+		log.Printf("Presigning failed: %s", err)
+		return
+	}
+
+	outDir, err := os.MkdirTemp(os.TempDir(), "thumb-*")
+	if err != nil {
+		log.Printf("Temp file creation failed: %s", err)
+		return
+	}
+	defer os.RemoveAll(outDir)
+	outFile := filepath.Join(outDir, "out.jpg")
+
+	args := []string{
+		"-i", presigned,
+		"-ss", "00:00:00",
+		"-vframes", "1",
+		"-y",
+		outFile,
+	}
+
+	timeout, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(timeout, "ffmpeg", args...)
+
+	var outputBuf bytes.Buffer
+	var stdErr bytes.Buffer
+	cmd.Stdout = &outputBuf
+	cmd.Stderr = &stdErr
+
+	err = cmd.Run()
+	if err != nil {
+		log.Printf("ffmpeg failed[%s] [%s]: %s", outputBuf.String(), stdErr.String(), err)
+		return
+	}
+
+	f, err := os.Open(outFile)
+	if err != nil {
+		log.Printf("Opening file failed: %s", err)
+		return
+	}
+	defer f.Close()
+	_, err = session.SaveData(context.Background(), "../latest.jpg", f, &drivers.FileProperties{CacheControl: "max-age=5"}, 1*time.Minute)
+	if err != nil {
+		log.Printf("Saving thumbnail failed: %s", err)
```

Comment:
Let's get some context on these log lines so we can correlate to a particular upload URL